connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfig.java

```diff
@@ -400,6 +412,53 @@ public WorkerConfig(ConfigDef definition, Map<String, String> props) {
         logInternalConverterDeprecationWarnings(props);
     }
 
+    @Override
+    public String toString() {
+        return "Comma-separated header rules, where each header rule is of the form "
+                + "'[action] [header name]:[header value]' and optionally surrounded by double quotes "
+                + "if any part of a header rule contains a comma";
+    }
+
+    public static void validateHttpResponseHeaderConfig(String config) {
+        try {
+            // validate format
+            String[] configTokens = config.trim().split("\\s+", 2);
+            if (configTokens.length != 2) {
+                throw new ConfigException(String.format("Invalid format of header config '%s\'. "
+                        + "Expected: '[ation] [header name]:[header value]'", config));
+            }
+
+            // validate action
+            String method = configTokens[0].trim();
+            validateHeaderConfigAction(method);
+
+            // validate header name and header value pair
+            String header = configTokens[1];
+            String[] headerTokens = header.trim().split(":");
+            if (headerTokens.length != 2) {
+                throw new ConfigException(
+                        String.format("Invalid format of header name and header value pair '%s'. "
+                                + "Expected: '[header name]:[header value]'", header));
+            }
+
+            // validate header name
+            String headerName = headerTokens[0].trim();
+            if (headerName.isEmpty() || headerName.matches(".*\\s+.*")) {
+                throw new ConfigException(String.format("Invalid header name '%s'. "
+                        + "The '[header name]' cannot contain whitespace", headerName));
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new ConfigException(String.format("Invalid header config '%s'.", config), e);
+        }
+    }
+
+    public static void validateHeaderConfigAction(String action) {
```

Comment:
This should be package-level protected:
```suggestion
    // Visible for testing
    static void validateHeaderConfigAction(String action) {
```